src/metabase/lib/temporal_bucket/util.cljc

```diff
@@ -30,26 +32,56 @@
     this-interval-message
     prev-interval-message
     next-interval-message]
+   (assert (every? string? [unit-message
+                            this-interval-message
+                            prev-interval-message
+                            next-interval-message])
+           "[[temporal-interval-tru]] accepts compile-time strings only.")
    `(temporal-interval-tru
      ~n
      ~this-interval-message
      ~prev-interval-message
-     ~(str "Previous {0} " unit-message)
      ~(str "Previous {0} " unit-message "s")
      ~next-interval-message
-     ~(str "Next {0} " unit-message)
      ~(str "Next {0} " unit-message "s")))
   ([n
     this-interval-message
     prev-interval-message
-    prev-single-interval-message
-    prev-plural-interval-message
+    prev-interval-plural-message
     next-interval-message
-    next-single-interval-message
-    next-plural-interval-message]
-   `(cond
-      (zero? ~n) (i18n/tru ~this-interval-message)
-      (= ~n -1)  (i18n/tru ~prev-interval-message)
-      (= ~n 1)   (i18n/tru ~next-interval-message)
-      (neg? ~n)  (i18n/trun ~prev-single-interval-message ~prev-plural-interval-message (abs ~n))
-      (pos? ~n)  (i18n/trun ~next-single-interval-message ~next-plural-interval-message ~n))))
+    next-interval-plural-message]
+   (assert (every? string? [this-interval-message
+                            prev-interval-message
+                            prev-interval-plural-message
+                            next-interval-message
+                            next-interval-plural-message])
+           "[[temporal-interval-tru]] accepts compile-time strings only.")
+   `(let [n# ~n]
+      (cond
+        (zero? n#) (i18n/tru ~this-interval-message)
+        (= n# -1)  (i18n/tru ~prev-interval-message)
+        (= n# 1)   (i18n/tru ~next-interval-message)
+        (neg? n#)  (i18n/trun ~prev-interval-message ~prev-interval-plural-message (abs n#))
+        (pos? n#)  (i18n/trun ~next-interval-message ~next-interval-plural-message n#)
```

Comment:
Will `prev-interval-message` and `next-interval-message` pass the validations done by `i18n/trun`?